### PR TITLE
feat(gatsby-source-store): Add source store plugin

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
Adds a new plugin called gatsby-source-store. This plugin should replace gatsby-source-vtex and works with store-api graphql schema. It sources both products and collections into the gatsby's graphql layer

### CODE REVIEW

File: packages/gatsby-source-store/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "test": "tsdx test",
+    "lint": "tsdx lint"
+  },
+  "peerDependencies": {
+    "gatsby": "^3.11.1",
+    "graphql": "^15.0.0"
+  },
+  "dependencies": {
+    "@graphql-tools/delegate": "^7.1.5",
+    "@graphql-tools/wrap": "^7.0.8",
```
Comment (emersonlaurentino) [inline packages/gatsby-source-store/package.json:36]: Do we need these dependencies?

File: packages/gatsby-source-store/src/sourceStore.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+} from 'gatsby-graphql-source-toolkit/dist/types'
+
+import type { Options } from './gatsby-node'
+import { RelayForward } from './paginate'
+
+export const sourceStoreType = async (
+  gatsbyApi: SourceNodesArgs,
+  options: Options,
+  gatsbyNodeTypes: IGatsbyNodeConfig[],
+  maxItems: number
```
Comment (emersonlaurentino) [inline packages/gatsby-source-store/src/sourceStore.ts:22]: what do you think about transforming all parameters into one object?

File: packages/gatsby-source-store/src/sourceStore.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        schema,
+      })
+
+      if (response.errors?.length) {
+        response.errors.forEach((e) => console.error(e))
+      }
+
+      return response
+    },
+    gatsbyTypePrefix: ``,
```
Comment (emersonlaurentino) [inline packages/gatsby-source-store/src/sourceStore.ts:54]: We can just remove that?
```suggestion
```

Comment (tlgimenes) [reply packages/gatsby-source-store/src/sourceStore.ts:54]: No, this is required for typescript not breaking

Comment (tlgimenes) [reply packages/gatsby-source-store/src/sourceStore.ts:54]: this property is required on the typings of this type

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**